### ISSUE TITLE
Paddle的set_default_dtype对BatchNorm不起作用，BatchNorm的参数仍然是float32

### ISSUE THREAD

**Issue by GuoQuanhao** [PaddlePaddle/Paddle#51708]
### bug描述 Describe the Bug

Paddle的set_default_dtype对BatchNorm不起作用，BatchNorm的参数仍然是float32
```python
import paddle
paddle.set_default_dtype("float16")
AA = paddle.nn.BatchNorm2D(10)
print(AA.parameters())
CC = paddle.nn.Linear(10,10)
print(CC.parameters())
```
```
[Parameter containing:
Tensor(shape=[10], dtype=float32, place=Place(gpu:0), stop_gradient=False,
       [1., 1., 1., 1., 1., 1., 1., 1., 1., 1.]), Parameter containing:
Tensor(shape=[10], dtype=float32, place=Place(gpu:0), stop_gradient=False,
       [0., 0., 0., 0., 0., 0., 0., 0., 0., 0.]), Parameter containing:
Tensor(shape=[10], dtype=float32, place=Place(gpu:0), stop_gradient=True,
       [0., 0., 0., 0., 0., 0., 0., 0., 0., 0.]), Parameter containing:
Tensor(shape=[10], dtype=float32, place=Place(gpu:0), stop_gradient=True,
       [1., 1., 1., 1., 1., 1., 1., 1., 1., 1.])]
[Parameter containing:
Tensor(shape=[10, 10], dtype=float16, place=Place(gpu:0), stop_gradient=False,
       [[ 0.05145,  0.37427, -0.32397, -0.47241, -0.45801, -0.31079,  0.29883, -0.20728,  0.47485,  0.49487],
        [-0.22241,  0.00028,  0.26587,  0.05484,  0.46899,  0.11121,  0.54639, -0.01548,  0.18823, -0.01753],
        [-0.36890, -0.13086, -0.36646, -0.25269, -0.25122, -0.43726, -0.13428, -0.40161, -0.17676, -0.13257],
        [-0.29517, -0.42627, -0.02448,  0.17834, -0.19580,  0.41699,  0.13672, -0.03757,  0.15100, -0.04517],
        [ 0.21094, -0.28149,  0.30664, -0.09900, -0.44946, -0.40210, -0.11786, -0.48926, -0.34448, -0.26318],
        [-0.03162,  0.40552,  0.31201,  0.25073,  0.50391,  0.13196, -0.31396,  0.13208,  0.41821,  0.40356],
        [-0.36035,  0.36084,  0.33008, -0.35474, -0.38843, -0.43262,  0.18457,  0.18628, -0.22449, -0.16833],
        [-0.11389, -0.46826, -0.27173, -0.07391,  0.11646,  0.02698,  0.39526,  0.26123, -0.35181, -0.09857],
        [-0.50732, -0.19519,  0.40088,  0.25610,  0.11053, -0.26782,  0.39819,  0.01814,  0.20874,  0.03671],
        [-0.22827, -0.24158,  0.16113, -0.10059, -0.10669, -0.23877,  0.41724, -0.37646, -0.27637,  0.24829]]), Parameter containing:
Tensor(shape=[10], dtype=float16, place=Place(gpu:0), stop_gradient=False,
       [0., 0., 0., 0., 0., 0., 0., 0., 0., 0.])]
```

**Comment by GuoQuanhao** [comment]
Paddle版本为2.4.2

**Comment by ceci3** [comment]
你好，感谢反馈，但是因为精度问题，batchnorm 是明确要求参数不能是fp16的，一般fp16输入都是用fp32的来表达参数，输入可以是fp16
